lib/matplotlib/colors.py

```diff
@@ -286,10 +286,25 @@ def to_rgba_array(c, alpha=None):
         return np.array([to_rgba(c, alpha)], float)
     except (ValueError, TypeError):
         pass
+
     # Convert one at a time.
-    result = np.empty((len(c), 4), float)
-    for i, cc in enumerate(c):
-        result[i] = to_rgba(cc, alpha)
+    if isinstance(c, str):
+        # Single string as color sequence.
+        # This is deprecated and will be removed in the future.
+        try:
+            result = np.array([to_rgba(cc, alpha) for cc in c], dtype=float)
```

Comment:
I think we don't need the `dtype=float` here and below?  to_rgba returns a tuple floats, so we're already good.

Reply:
I thought I'd leave it just to be explicit. But you are right, it's not needed as `to_rgba` *should* return floats (which is not documented and not easy to see from the code). If you want, I can remove it and rely on `to_rgba` not doing anything bad.

Reply:
Would rather remove it (and document to_rgba if you want), but not insisting on that.

Reply:
removed.